### PR TITLE
Remove unused code

### PR DESCRIPTION
**Pull request by stweil**
Several source files were removed because they only contained unused code.

### CODE REVIEW

**Comment by stweil** [comment]
The unused code was found by comparing the symbol names of the built Tesseract library and executable using a static build.

**Comment by stweil** [comment]
The Tesseract library still contains lots of symbols which are only needed for the training programs:

    tesseract::CTC::Backward(GENERIC_2D_ARRAY<double>*) const
    tesseract::CTC::CalculateBiasFraction()
    tesseract::CTC::ComputeCTCTargets(GenericVector<int> const&, int, GENERIC_2D_ARRAY<float> const&, tesseract::NetworkIO*)
    tesseract::CTC::ComputeLabelLimits()
    tesseract::CTC::ComputeSimpleTargets(GENERIC_2D_ARRAY<float>*) const
    tesseract::CTC::ComputeWidthsAndMeans(GenericVector<float>*, GenericVector<int>*) const
    tesseract::CTC::CTC(GenericVector<int> const&, int, GENERIC_2D_ARRAY<float> const&)
    tesseract::CTC::CTC(GenericVector<int> const&, int, GENERIC_2D_ARRAY<float> const&)
    tesseract::CTC::Forward(GENERIC_2D_ARRAY<double>*) const
    tesseract::CTC::LabelsToClasses(GENERIC_2D_ARRAY<double> const&, tesseract::NetworkIO*) const
    tesseract::CTC::NeededNull(int) const
    tesseract::CTC::NormalizeProbs(GENERIC_2D_ARRAY<float>*)
    tesseract::CTC::NormalizeSequence(GENERIC_2D_ARRAY<double>*) const
    tesseract::ErrorCounter::AccumulateErrors(bool, tesseract::CountTypes, tesseract::FontInfoTable const&, GenericVector<tesseract::UnicharRating> const&, tesseract::TrainingSample*)
    tesseract::ErrorCounter::AccumulateJunk(bool, GenericVector<tesseract::UnicharRating> const&, tesseract::TrainingSample*)
    tesseract::ErrorCounter::ComputeErrorRate(tesseract::ShapeClassifier*, int, tesseract::CountTypes, tesseract::FontInfoTable const&, GenericVector<Pix*> const&, tesseract::SampleIterator*, double*, double*, STRING*)
    tesseract::ErrorCounter::ComputeRates(tesseract::ErrorCounter::Counts const&, double*)
    tesseract::ErrorCounter::Counts::Counts()
    tesseract::ErrorCounter::Counts::Counts()
    tesseract::ErrorCounter::Counts::operator+=(tesseract::ErrorCounter::Counts const&)
    tesseract::ErrorCounter::DebugNewErrors(tesseract::ShapeClassifier*, tesseract::ShapeClassifier*, tesseract::CountTypes, tesseract::FontInfoTable const&, GenericVector<Pix*> const&, tesseract::SampleIterator*)
    tesseract::ErrorCounter::ErrorCounter(UNICHARSET const&, int)
    tesseract::ErrorCounter::ErrorCounter(UNICHARSET const&, int)
    tesseract::ErrorCounter::ReportErrors(int, tesseract::CountTypes, tesseract::FontInfoTable const&, tesseract::SampleIterator const&, double*, STRING*)
    tesseract::ErrorCounter::ReportString(bool, tesseract::ErrorCounter::Counts const&, STRING*)
    tesseract::IntFeatureDist::Clear()
    tesseract::IntFeatureDist::DebugFeatureDistance(GenericVector<int> const&) const
    tesseract::IntFeatureDist::FeatureDistance(GenericVector<int> const&) const
    tesseract::IntFeatureDist::Init(tesseract::IntFeatureMap const*)
    tesseract::IntFeatureDist::~IntFeatureDist()
    tesseract::IntFeatureDist::~IntFeatureDist()
    tesseract::IntFeatureDist::IntFeatureDist()
    tesseract::IntFeatureDist::IntFeatureDist()
    tesseract::IntFeatureDist::Set(GenericVector<int> const&, int, bool)
    tesseract::IntFeatureMap::Clear()
    tesseract::IntFeatureMap::ComputeOffsetFeature(int, int) const
    tesseract::IntFeatureMap::DebugMapFeatures(GenericVector<int> const&) const
    tesseract::IntFeatureMap::DeleteMapFeature(int)
    tesseract::IntFeatureMap::FinalizeMapping(tesseract::SampleIterator*)
    tesseract::IntFeatureMap::IndexFeature(INT_FEATURE_STRUCT const&) const
    tesseract::IntFeatureMap::Init(tesseract::IntFeatureSpace const&)
    tesseract::IntFeatureMap::~IntFeatureMap()
    tesseract::IntFeatureMap::~IntFeatureMap()
    tesseract::IntFeatureMap::IntFeatureMap()
    tesseract::IntFeatureMap::IntFeatureMap()
    tesseract::IntFeatureMap::InverseIndexFeature(int) const
    tesseract::IntFeatureMap::InverseMapFeature(int) const
    tesseract::IntFeatureMap::IsMapFeatureDeleted(int) const
    tesseract::IntFeatureMap::MapFeature(INT_FEATURE_STRUCT const&) const
    tesseract::IntFeatureMap::MapIndexFeature(int) const
    tesseract::IntFeatureMap::OffsetFeature(int, int) const
    tesseract::MasterTrainer::AddSample(bool, char const*, tesseract::TrainingSample*)
    tesseract::MasterTrainer::AddSpacingInfo(char const*)
    tesseract::MasterTrainer::ClusterShapes(int, int, float, tesseract::ShapeTable*)
    tesseract::MasterTrainer::DebugCanonical(char const*, char const*)
    tesseract::MasterTrainer::DisplaySamples(char const*, int, char const*, int)
    tesseract::MasterTrainer::GetBestMatchingFontInfoId(char const*)
    tesseract::MasterTrainer::GetFontInfoId(char const*)
    tesseract::MasterTrainer::IncludeJunk()
    tesseract::MasterTrainer::LoadFontInfo(char const*)
    tesseract::MasterTrainer::LoadPageImages(char const*)
    tesseract::MasterTrainer::LoadUnicharset(char const*)
    tesseract::MasterTrainer::LoadXHeights(char const*)
    tesseract::MasterTrainer::~MasterTrainer()
    tesseract::MasterTrainer::~MasterTrainer()
    tesseract::MasterTrainer::MasterTrainer(tesseract::NormalizationMode, bool, bool, int)
    tesseract::MasterTrainer::MasterTrainer(tesseract::NormalizationMode, bool, bool, int)
    tesseract::MasterTrainer::PostLoadCleanup()
    tesseract::MasterTrainer::PreTrainingSetup()
    tesseract::MasterTrainer::ReadTrainingSamples(char const*, FEATURE_DEFS_STRUCT const&, bool)
    tesseract::MasterTrainer::ReplaceFragmentedSamples()
    tesseract::MasterTrainer::ReplicateAndRandomizeSamplesIfRequired()
    tesseract::MasterTrainer::Serialize(_IO_FILE*) const
    tesseract::MasterTrainer::SetupFlatShapeTable(tesseract::ShapeTable*)
    tesseract::MasterTrainer::SetupForClustering(tesseract::ShapeTable const&, FEATURE_DEFS_STRUCT const&, int, int*)
    tesseract::MasterTrainer::SetupMasterShapes()
    tesseract::MasterTrainer::ShapeDistance(tesseract::ShapeTable const&, int, int)
    tesseract::MasterTrainer::TestClassifierOnSamples(tesseract::CountTypes, int, bool, tesseract::ShapeClassifier*, STRING*)
    tesseract::MasterTrainer::TestClassifier(tesseract::CountTypes, int, bool, tesseract::TrainingSampleSet*, tesseract::ShapeClassifier*, STRING*)
    tesseract::MasterTrainer::TestClassifierVOld(bool, tesseract::ShapeClassifier*, tesseract::ShapeClassifier*)
    tesseract::MasterTrainer::WriteInttempAndPFFMTable(UNICHARSET const&, UNICHARSET const&, tesseract::ShapeTable const&, CLASS_STRUCT*, char const*, char const*)
    tesseract::NetworkBuilder::BuildFromString(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::BuildLSTMXYQuad(int, int)
    tesseract::NetworkBuilder::InitNetwork(int, STRING, int, int, float, tesseract::TRand*, tesseract::Network**)
    tesseract::NetworkBuilder::ParseC(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseFullyConnected(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseInput(char**)
    tesseract::NetworkBuilder::ParseLSTM(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseM(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseOutput(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseParallel(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseR(tesseract::StaticShape const&, char**)
    tesseract::NetworkBuilder::ParseSeries(tesseract::StaticShape const&, tesseract::Input*, char**)
    tesseract::NetworkBuilder::ParseS(tesseract::StaticShape const&, char**)
    tesseract::SampleIterator::AtEnd() const
    tesseract::SampleIterator::Begin()
    tesseract::SampleIterator::Clear()
    tesseract::SampleIterator::CompactCharsetSize() const
    tesseract::SampleIterator::GetCompactClassID() const
    tesseract::SampleIterator::GetSample() const
    tesseract::SampleIterator::GetShapeEntry() const
    tesseract::SampleIterator::GetSparseClassID() const
    tesseract::SampleIterator::GlobalSampleIndex() const
    tesseract::SampleIterator::Init(tesseract::IndexMapBiDi const*, tesseract::ShapeTable const*, bool, tesseract::TrainingSampleSet*)
    tesseract::SampleIterator::MapSampleFeatures(tesseract::IntFeatureMap const&)
    tesseract::SampleIterator::MutableSample() const
    tesseract::SampleIterator::Next()
    tesseract::SampleIterator::NormalizeSamples()
    tesseract::SampleIterator::~SampleIterator()
    tesseract::SampleIterator::~SampleIterator()
    tesseract::SampleIterator::SampleIterator()
    tesseract::SampleIterator::SampleIterator()
    tesseract::SampleIterator::SparseCharsetSize() const
    tesseract::SampleIterator::UniformSamples()
    tesseract::TrainingSampleSet::AddAllFontsForClass(int, tesseract::Shape*) const
    tesseract::TrainingSampleSet::AddSample(char const*, tesseract::TrainingSample*)
    tesseract::TrainingSampleSet::AddSample(int, tesseract::TrainingSample*)
    tesseract::TrainingSampleSet::ClusterDistance(int, int, int, int, tesseract::IntFeatureMap const&)
    tesseract::TrainingSampleSet::ComputeCanonicalFeatures()
    tesseract::TrainingSampleSet::ComputeCanonicalSamples(tesseract::IntFeatureMap const&, bool)
    tesseract::TrainingSampleSet::ComputeCloudFeatures(int)
    tesseract::TrainingSampleSet::ComputeClusterDistance(int, int, int, int, tesseract::IntFeatureMap const&) const
    tesseract::TrainingSampleSet::DeleteableSample(tesseract::TrainingSample const*)
    tesseract::TrainingSampleSet::DeleteDeadSamples()
    tesseract::TrainingSampleSet::DeSerialize(bool, _IO_FILE*)
    tesseract::TrainingSampleSet::DisplaySamplesWithFeature(int, tesseract::Shape const&, tesseract::IntFeatureSpace const&, ScrollView::Color, ScrollView*) const
    tesseract::TrainingSampleSet::FontClassInfo::DeSerialize(bool, _IO_FILE*)
    tesseract::TrainingSampleSet::FontClassInfo::FontClassInfo()
    tesseract::TrainingSampleSet::FontClassInfo::FontClassInfo()
    tesseract::TrainingSampleSet::FontClassInfo::Serialize(_IO_FILE*) const
    tesseract::TrainingSampleSet::GetCanonicalDist(int, int) const
    tesseract::TrainingSampleSet::GetCanonicalFeatures(int, int) const
    tesseract::TrainingSampleSet::GetCanonicalSample(int, int) const
    tesseract::TrainingSampleSet::GetCloudFeatures(int, int) const
    tesseract::TrainingSampleSet::GetSample(int) const
    tesseract::TrainingSampleSet::GetSample(int, int, int) const
    tesseract::TrainingSampleSet::GlobalSampleIndex(int, int, int) const
    tesseract::TrainingSampleSet::IndexFeatures(tesseract::IntFeatureSpace const&)
    tesseract::TrainingSampleSet::KillSample(tesseract::TrainingSample*)
    tesseract::TrainingSampleSet::LoadUnicharset(char const*)
    tesseract::TrainingSampleSet::MutableSample(int, int, int)
    tesseract::TrainingSampleSet::NumClassSamples(int, int, bool) const
    tesseract::TrainingSampleSet::OrganizeByFontAndClass()
    tesseract::TrainingSampleSet::ReliablySeparable(int, int, int, int, tesseract::IntFeatureMap const&, bool) const
    tesseract::TrainingSampleSet::ReplicateAndRandomizeSamples()
    tesseract::TrainingSampleSet::SampleToString(tesseract::TrainingSample const&) const
    tesseract::TrainingSampleSet::Serialize(_IO_FILE*) const
    tesseract::TrainingSampleSet::SetupFontIdMap()
    tesseract::TrainingSampleSet::~TrainingSampleSet()
    tesseract::TrainingSampleSet::~TrainingSampleSet()
    tesseract::TrainingSampleSet::TrainingSampleSet(tesseract::FontInfoTable const&)
    tesseract::TrainingSampleSet::TrainingSampleSet(tesseract::FontInfoTable const&)
    tesseract::TrainingSampleSet::UnicharDistance(tesseract::UnicharAndFonts const&, tesseract::UnicharAndFonts const&, bool, tesseract::IntFeatureMap const&)

**Comment by stweil** [comment]
Pull request #2614 removes the above symbols from the Tesseract library by moving code from `src/classify` and `src/lstm` to `src/training`.
